### PR TITLE
Deep link to add secondary login in OldDot

### PR DESCRIPTION
**Pull request by luacmartins**
This PR is a copy of https://github.com/Expensify/App/pull/6176. I forgot to add a hold on the previous PR and it was merged before it should have been and we reverted it.

cc @mountiny 

### Details
Copy changes to `Issue corporate cards` and `Connect bank account` page and deep-linking to `Add secondary login` in OldDot. 

### Fixed Issues
https://github.com/Expensify/Expensify/issues/182071 (partial)

### Tests
1. Create an account using a public domain, e.g `@gmail.com`
2. Follow the steps in [this SO](https://stackoverflow.com/c/expensify/questions/342) to add an "OPEN" bank account.
3. When you reach the last step, verify the following:
- **Copy** 
>This bank account will be used to reimburse expenses, collect invoices, and pay bills all from the same account. Please add a work email address as a secondary login to enable the Expensify Card.
- **Button** 
>Add work email address
4. Click the button and verify that it opens new browser tab on OldDot and directs user to `Settings > Account Details > Add secondary login` with the secondary login modal open.
5. Go back to NewDot and navigate to `Settings > Select Workspace > Issue corporate cards`.
6. Verify that the following:
- **Copy**
>Add a work email address to issue unlimited Expensify Cards for your workspace members, as well as all of these incredible benefits:
 
>Up to 2% cash back
Digital and physical cards
No personal liability
Customizable limits
- **Button**
>Add work email address
4. Click the button and verify that it opens new browser tab on OldDot and directs user to `Settings > Account Details > Add secondary login` with the secondary login modal open.

### QA Steps
Steps above.

### Tested On

- [X] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
web-1

web-2

#### Mobile Web

mweb-1

mweb-2

#### Desktop

desktop-1

desktop-2

#### iOS
ios-1

ios-2

#### Android
android-1

andorid-2

### CODE REVIEW

**Review by mountiny — DISMISSED** [review]
Approved it once, gonna approve it again 😅

**Review by Jag96 — COMMENTED** [review]
A couple of questions, otherwise looks good!

File: src/libs/actions/User.js
Comment (luacmartins) [reply src/libs/actions/User.js:291]: This is a one-time only subscription that happens when the user clicks the `Add work email address` link. Adding it to `subscribeToUserEvents` would subscribe users prematurely and unnecessarily.

I didn't add the check because when the user subscribes to this event we should have a `currentUserAccountID`. That is different from `subscribeToUserEvents` because `subscribeToUserEvents` is called upon login and race conditions could cause `currentUserAccountID` to be undefined. With that being said, I'll add the check just to be safe.

File: src/libs/actions/User.js
Comment (luacmartins) [reply src/libs/actions/User.js:296]: Setting it to `null` should delete the key from Onyx. Since this is a one-time only event, I felt like it was better to remove the key than leave it set as `false`.

File: src/libs/actions/User.js
Comment (Jag96) [reply src/libs/actions/User.js:295]: Sounds good, thanks for the explanation!

**Review by Jag96 — APPROVED** [review]

**Review by mountiny — APPROVED** [review]

**Comment by luacmartins** [comment]
No longer on hold! Self-merging since it was approved.

**Comment by isagoico** [comment]
We found this issue when retesting this on Android https://github.com/Expensify/App/issues/6457. I don't think this PR is the culprit do so checking it off. Tested this on Android via adding the bank account through web and then logging into that same account in Android. 
image

**Comment by isagoico** [comment]
@luacmartins @Jag96 Question about the OldDot redirection. This is expected to open this mWeb page right? I ask because OldDot is messy sometimes in mWeb (in this case it looks ok but I wanted to confirm if this was the expected behaviour)

**Comment by isagoico** [comment]
@luacmartins Hello! This issue was found while executing this PR in iOS https://github.com/Expensify/App/issues/6475 and it's currently failing the  - It's a different behaviour from what we have in Android. Can you give it a look 🙇‍♀️

**Comment by luacmartins** [comment]
Hi @isagoico! I put up a fix for it https://github.com/Expensify/App/pull/6479. Thanks for bringing this up to my attention!

**Comment by mananjadhav** [comment]
I am tagging this PR to highlight an issue fixed [here](https://github.com/Expensify/App/issues/16683). All conditions in ternary expressions or left-hand operands on conditional renders, should be boolean. This PR is one of the PRs that uses conditional render with string operands, hence I am tagging it here for the contributors to check. 

We've also updated the item in the checklist with this [PR](https://github.com/Expensify/App/pull/18069) to avoid this issue in the future.

**Pull request merged**
